Felienne/hedy
# [FEATURE] Enables filtering on programs page

**Description**
In this PR we implement filtering on the programs page. Similarly to how the `/explore` page works we add a search header for filtering on level and adventure. Using a GET these attributes are sent to the back-end and the page is reloaded with the corresponding programs. We also fix an important bug where a teacher viewing a students page was able to set favorite programs for the public profile. (Not really because we perform an additional check on the back-end if the current user is owner of that program, but still).

**Fix for**
This PR fixes #1678.

**How to test**
Navigate to the `programs` page and notice the new header. Try different filtering options as user as well as teacher of a user (when viewing a student's profile). Make sure that everything works as expected.
